controller/plan_controller.py

```diff
@@ -89,8 +94,11 @@ def close_plan(plan: Plan):
     Use find_plan in combination with this function when admin requests for a plan to be closed.
     Inputted plan will be changed to read-only by changing the __is_closed flag in Plan class.
     """
-    plan.close()
-    plan.save()
+    if plan.is_closed:
+        raise controller_error.ControllerError("Plan is already closed.")
```

Comment:
This is optional but I think it might be clearer to show:
`raise controller_error.ControllerError(f"Plan {plan.name} is already closed.")`

Reply:
Makes sense, I've changed it now!